crypto/stateproof/builder.go

```diff
@@ -46,6 +46,8 @@ type Builder struct {
 	lnProvenWeight uint64
 	provenWeight   uint64
 	strengthTarget uint64
+	stateProof     *StateProof
```

Comment:
1 should that be `cachedStateProof`?
2 why don't we give up skipBuild and make it a nil ptr when `Add` is invoked?